src/docs/eviction.dox

```diff
@@ -0,0 +1,103 @@
+committed value of each key to the disk image. All the older versions
+are moved to the history store table. The new disk image is
+subsequently flushed to the data file on the disk and the page is
+removed from memory.
+
+The history store table is the internal table for WiredTiger to store
+historical versions for all the user-created tables. The pages in the
+history store table need to be evicted and loaded back to memory as
+well. Each entry in the history store represents a historical value
+of a key in the user-created tables. Different from the user-created
+tables, all the content on the history store page is committed as the
+historical values it stores must all be committed, and each entry only
+has one value and a deletion marking the expiration of the entry on
+its update chain. Therefore, history store pages should always be clean
+after reconciliation and can be evicted from memory. Besides, it
+prevents eviction from moving content in the history store recursively
+to the history store during reconciliation as keys in the history store
+always only have one committed value.
+
+@section update_restore_eviction Update restore eviction
+
+Update restore eviction largely overlaps with history store eviction.
+The difference is that there are uncommitted changes on the page and
+WiredTiger cannot write them to the disk image. Instead, besides
+writing the disk image to the disk, eviction also keeps the disk image
+in memory and restores the uncommitted changes to the new disk image.
+
+Since usually there is only a subset of keys that have uncommitted
+changes, eviction only needs to restore those keys that are dirty
+after reconciliation. Apart from that, eviction may split a page into
+multiple pages and some of them may be clean after reconciliation
+even though the old page before split has uncommitted changes.
+Therefore, WiredTiger tracks whether it needs to do update restore
+eviction at the key level and the page level.
+
+At the key level, the key needs to be restored if it has uncommitted
+updates. For each new page, restoration is required if any of the
+keys on the new page needs to be restored. Once eviction decides to
+restore a page, it will make a copy of the disk image in memory
+during reconciliation.
+
+For the restored keys, eviction also frees all the values that have
+been written to the disk image or moved to the history store as there
+is no need to duplicate them on the update chains. For the clean keys
+that are not restored, their whole update chains are removed from
+memory along with the old disk image.
+
+@section exceptions Exceptions
+
+In-memory database and fixed-length column store are the exceptions.
+Since they don't support the history store, eviction can only discard
+the update chains until all the content on the update chains are
+globally visible. For the same reason, we cannot free the updates
+older than the update written to the disk image. Also for in-memory
```

Comment:
Consider rewording the last sentence:  "For in-memory databases, eviction always copies the new disk image and keeps it in memory as there is no disk storage."

Reply:
Done